rotkehlchen/chain/zksync_lite/manager.py

```diff
@@ -165,26 +165,28 @@ def _query_api(
 
         return result
 
-    def _query_and_save_transactions_for_range(
+    def _query_and_save_transactions_from_hash(
             self,
             address: ChecksumEvmAddress,
-            start_ts: Timestamp,
-            end_ts: Timestamp,
             from_hash: str,
             direction: Literal['older', 'newer'],
     ) -> None:
-        """Save transactions in a timerange. Timerange is not really respected.
-        Just saved in the DB range."""
-        ranges = DBQueryRanges(self.database)
-        location = f'{ZKSYNCLITE_TX_SAVEPREFIX}{address}'
-        current_start_ts = start_ts
-        current_end_ts = end_ts
+        """Query and save transactions before or since the transaction specified with from_hash."""
         input_transactions: set[ZKSyncLiteTransaction] = set()
         for new_transactions in self._query_zksync_api_transactions(
                 address=address,
                 from_hash=from_hash,
                 direction=direction,
         ):
+            if (
+                from_hash != 'latest' and
+                len(new_transactions) != 0 and
+                new_transactions[0].tx_hash.hex() == from_hash
+            ):
+                # Remove first transaction if it matches from_hash
```

Comment:
why would the first transaction match and already be in the DB? This seems like a pretty random thing to check for?

Reply:
Added a comment in the code to explain this.